iothub/service/tests/ExceptionHandlingHelperTests.cs

```diff
@@ -29,17 +31,36 @@ public async Task GetExceptionCodeAsync_ContentAndHeadersMatch_ValidErrorCode()
             httpResponseMessage.Headers.Add(HttpErrorCodeName, "DeviceNotFound");
 
             // act
-            ErrorCode errorCode = await ExceptionHandlingHelper.GetExceptionCodeAsync(httpResponseMessage);
+            IotHubErrorCode errorCode = await ExceptionHandlingHelper.GetIotHubErrorCodeAsync(httpResponseMessage);
 
             // assert
-            Assert.AreEqual(ErrorCode.DeviceNotFound, errorCode);
+            errorCode.Should().Be(IotHubErrorCode.DeviceNotFound);
         }
 
         [TestMethod]
-        public async Task GetExceptionCodeAsync_ContentAndHeadersMisMatch_InvalidErrorCode()
+        public async Task GetExceptionCodeAsync_ContentAndHeadersMatch_NonNumericErrorCode_ValidErrorCode()
         {
             // arrange
-            var httpResponseMessage = new HttpResponseMessage(System.Net.HttpStatusCode.BadRequest);
+            var httpResponseMessage = new HttpResponseMessage(HttpStatusCode.BadRequest);
+            var exceptionResult = new IoTHubExceptionResult
+            {
+                Message = "errorCode:PreconditionFailed"
+            };
+            httpResponseMessage.Content = new StringContent(JsonConvert.SerializeObject(exceptionResult));
+            httpResponseMessage.Headers.Add(HttpErrorCodeName, "PreconditionFailed");
+
+            // act
+            IotHubErrorCode errorCode = await ExceptionHandlingHelper.GetIotHubErrorCodeAsync(httpResponseMessage);
+
+            // assert
+            errorCode.Should().Be(IotHubErrorCode.PreconditionFailed);
+        }
+
+        [TestMethod]
+        public async Task GetExceptionCodeAsync_ContentAndHeadersMisMatch_UnknownErrorCode()
```

Comment:
Mismatch one word, so no second capital M.

```suggestion
        public async Task GetExceptionCodeAsync_ContentAndHeadersMismatch_UnknownErrorCode()
```